### PR TITLE
Package wide options

### PR DESCRIPTION
**Pull request by jochemb**
Introduces option for backward compatibility with transmart-batch ( #32 )

### CODE REVIEW

File: tests/batch_tests.py [hunk 1]
```diff
@@ -1,29 +1,40 @@
...
+batch.jdbc.user=tm_cz
+batch.jdbc.password=tm_cz
+"""
 
-class BatchTests(unittest.TestCase):
-    @classmethod
-    def setUpClass(cls):
-        cls.study = tmtk.Study('studies/valid_study/study.params')
 
+class BatchTests(TestBase):
```
Comment (wibo-pipping) [inline tests/batch_tests.py]: I'm not sure we should test this function by default. This test will also fail if you have no transmart installation on your machine. Using python unittest you could extent the TestBase to include TestBaseEssential and TestBaseExtended?

Comment (jochemb) [reply tests/batch_tests.py]: They are setup to not fail if you don't have an installation. The tests are only for the wrapper that calls the subprocess and for the cli.

File: tests/commons.py [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+import tempfile
+
+import tmtk
+
+
+def create_study_from_dir(dir_name):
+    return tmtk.Study(os.path.join(TestBase.studies_dir, dir_name, 'study.params'))
+
+
+class TestBase(unittest.TestCase):
```
Comment (wibo-pipping) [inline tests/commons.py]: Consider extending the TestBase with sub classes to create separate suites. Examples could be Essentials and Extras. Maybe split out the tests to test smaller units, where the units could be Arborist, Batch, Skinny_Export etc.

File: tests/incomplete_studies_tests.py [hunk 3]
```diff
@@ -1,42 +1,30 @@
...
     def test_non_existing_wordmap(self):
         self.study.Params.clinical.WORD_MAP_FILE = 'fake_wordmap.txt'
-        assert self.study.Clinical.WordMapping.validate()
+        self.assertTrue(self.study.Clinical.WordMapping.validate())
         self.study.Params.clinical.WORD_MAP_FILE = 'Cell-line_wordmap.txt'
 
-    def test_create_json_string(self):
-        assert tmtk.arborist.create_concept_tree(self.study)
-
     def test_update_study_from_json(self):
```
Comment (wibo-pipping) [inline tests/incomplete_studies_tests.py]: What is this test asserting?

File: tmtk/options.py [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+
+
+transmart_batch_home_doc = """
+Path to transmart-batch home directory that contains the .jar file and 
+property files. By default this is the $TMBATCH_HOME environment variable.
+"""
+register_option('transmart_batch_home',
+                default=os.environ.get('TMBATCH_HOME', ''),
+                doc=transmart_batch_home_doc,
+                validator=is_str)
```
Comment (wibo-pipping) [inline tmtk/options.py]: Extend the validator to check a path as well.

Comment (jochemb) [reply tmtk/options.py]: nice idea, but too strict for my taste ;)

File: tests/incomplete_studies_tests.py (hunk 3 above)
Comment (jochemb) [reply tests/incomplete_studies_tests.py]: nothing, but running the code. I removed it cause it was duplicated.

File: tests/commons.py (hunk 2 above)
Comment (jochemb) [reply tests/commons.py]: Yes. good idea. Maybe for later?

**Pull request merged**